src/lib/containers/entity_finder/details/view/DetailsView.tsx

```diff
@@ -241,7 +237,10 @@ export const DetailsView: React.FunctionComponent<DetailsViewProps> = ({
         const entityType = getEntityTypeFromHeader(entity)
         entities.push({
           ...entity,
-          entityType,
+          entityId: entity.id,
+          versionNumber:
+            'versionNumber' in entity ? entity.versionNumber : undefined,
```

Comment:
Sharing renderers between the Entity Finder and the new dataset editor is easier when explicitly requiring both to supply `entityId` and `versionNumber`